### PR TITLE
Implement helm chart and skaffold setup

### PR DESCRIPTION
**Pull request by dimityrmirchev**
**What this PR does / why we need it**:
Implements helm chart and skaffold setup.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```feature user
NONE
```

### CODE REVIEW

File: charts/discovery-server/charts/runtime/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+            path: /healthz
+            port: healthz
+            scheme: HTTP
+          initialDelaySeconds: 5
+          timeoutSeconds: 5
+          failureThreshold: 3
+          successThreshold: 1
+          periodSeconds: 10
+        ports:
+        - name: http
```
Comment (vpnachev) [inline charts/discovery-server/charts/runtime/templates/deployment.yaml:79]:
```suggestion
        - name: https
```

File: charts/discovery-server/charts/runtime/templates/service.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  labels:
+{{ include "labels" . | indent 4 }}
+  annotations:
+    networking.resources.gardener.cloud/from-all-garden-scrape-targets-allowed-ports: '[{"protocol":"TCP","port":8080}]'
+spec:
+  type: ClusterIP
+  selector:
+{{ include "labels" . | indent 4 }}
+  ports:
+  - name: http
```
Comment (vpnachev) [inline charts/discovery-server/charts/runtime/templates/service.yaml:19]:
```suggestion
  - name: https
```

**Review by vpnachev — CHANGES_REQUESTED** [review]
Very nice, just some minor nits.

File: charts/discovery-server/charts/runtime/templates/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+  revisionHistoryLimit: 2
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+{{ include "labels" . | indent 6 }}
+  template:
+    metadata:
+      annotations:
+        {{- if .Values.kubeconfig }}
+        checksum/gardener-discovery-server-kubeconfig: {{ include (print $.Template.BasePath "/secret-kubeconfig.yaml") . | sha256sum }}
```
[the remark this answers is not in the document]
Comment (dimityrmirchev) [reply charts/discovery-server/charts/runtime/templates/deployment.yaml:23]: Whoops. I somehow forgot that. Thanks for noticing

**Pull request merged**